python/lib/client/dmod/client/dmod_client.py

```diff
@@ -64,6 +64,62 @@ def __init__(self, client_config: ClientConfig, bypass_request_service: bool = F
 
         self._auth_client: AuthClient = AuthClient(transport_client=self._get_transport_client())
 
+    def _extract_dataset_domain(self, **kwargs) -> DataDomain:
```

Comment:
Thoughts on making this a `staticmethod` just to make it clear that it does not modify the state of the instance, but makes sense to have it here for cohesion reasons?

Reply:
Hmm, it's an interesting thought exercise ... I didn't really intend for things to make it possible to call this outside an instance; I just wanted to tidy up the code a bit.  On the other hand, aside from implying no state changes, it's probably slightly more performant.

Rather than fall all the way down the rabbit hole, I went ahead and made the change.